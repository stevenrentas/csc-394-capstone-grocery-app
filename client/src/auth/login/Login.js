import {
  Typography,
  Box,
  TextField,
  Stack,
  FormControl,
  InputLabel,
  OutlinedInput,
  InputAdornment,
  IconButton,
  Button,
} from "@mui/material";
import Visibility from "@mui/icons-material/Visibility";
import VisibilityOff from "@mui/icons-material/VisibilityOff";
import React from "react";
import { useNavigate } from "react-router-dom";

const Login = () => {
  const baseURL = "http://localhost:3001/api/v1/users";
  const axios = require("axios");
  const loginAPI = "/auth/api/v1/users";
  const navigate = useNavigate();
  const submitLoginForm = async () => {
    axios
      .post(baseURL, {
        username: username,
        password: password,
      })
      .then((response) => {
        console.log(response);
        if (!response.data.isAuthed) {
          alert("Unable to login. Please try after some time.");
          return;
        }
        localStorage.clear();
        localStorage.setItem("user-token", response.data.token);
        setTimeout(() => {
          navigate("/");
        }, 500);
      })
      .catch((error) => {
        console.log(error);
      });
  };

  const [username, setUsername] = React.useState("");
  const [password, setPassword] = React.useState("");

  const [showPassword, setShowPassword] = React.useState(false);

  const handleClickShowPassword = () => setShowPassword((show) => !show);

  const handleMouseDownPassword = (event) => {
    event.preventDefault();
  };

  return (
    <Box sx={{ m: 5 }}>
      <Typography>
        Welcome to our grocery list app! Please enter your credentials below.
      </Typography>
      <Stack>
        <TextField
          id="username"
          label="Username"
          variant="outlined"
          sx={{ mt: 2, maxWidth: "200px" }}
          onChange={(e) => setUsername(e.target.value)}
        />
        <FormControl sx={{ maxWidth: "200px", mt: 2 }} variant="outlined">
          <InputLabel htmlFor="outlined-adornment-password">
            Password
          </InputLabel>
          <OutlinedInput
            id="outlined-adornment-password"
            onChange={(e) => setPassword(e.target.value)}
            type={showPassword ? "text" : "password"}
            endAdornment={
              <InputAdornment position="end">
                <IconButton
                  aria-label="toggle password visibility"
                  onClick={handleClickShowPassword}
                  onMouseDown={handleMouseDownPassword}
                  edge="end"
                >
                  {showPassword ? <VisibilityOff /> : <Visibility />}
                </IconButton>
              </InputAdornment>
            }
            label="Password"
          />
        </FormControl>
      </Stack>
      <Button sx={{ mt: 2 }} variant="contained" onClick={submitLoginForm}>
        Submit
      </Button>
    </Box>
  );
};

export default Login;
