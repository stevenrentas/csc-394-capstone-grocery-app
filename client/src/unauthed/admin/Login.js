import {
  Typography,
  Box,
  TextField,
  Stack,
  FormControl,
  InputLabel,
  OutlinedInput,
  InputAdornment,
  IconButton,
  Button,
} from "@mui/material";
import Visibility from "@mui/icons-material/Visibility";
import VisibilityOff from "@mui/icons-material/VisibilityOff";
import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import config from "../../api/api";
import { createTheme, ThemeProvider } from "@mui/material/styles";
import FilledInput from "@mui/material/FilledInput";

const Login = () => {
  const axios = require("axios");
  const api = axios.create({
    baseURL: config,
  });
  const navigate = useNavigate();
  const submitLoginForm = async () => {
    api
      .post("/users", {
        username: credentials.username,
        pword: credentials.password,
      })
      .then((response) => {
        if (response.data.isadmin !== true) {
          setError({
            message: "User does not have required access",
          });
          return;
        } else {
          localStorage.clear();
          localStorage.setItem("admin-token", response.data.token);
          localStorage.setItem("user-id", response.data.uid);
          setTimeout(() => {
            navigate("/admin/portal");
          }, 500);
        }
      })
      .catch((error) => {
        console.log(error);
      });
  };

  const theme = createTheme({
    palette: {
      primary: {
        main: "#698669",
        hover: "#283593", // Custom hover color for the primary button
      },
      secondary: {
        main: "#fffff0",
        hover: "#283593", // Custom hover color for the primary button
      },
    },
  });

  const [username, setUsername] = React.useState("");
  const [password, setPassword] = React.useState("");

  const [showPassword, setShowPassword] = React.useState(false);

  const handleClickShowPassword = () => setShowPassword((show) => !show);

  const handleMouseDownPassword = (event) => {
    event.preventDefault();
  };

  /* New */
  const [credentials, setCredentials] = useState({
    username: "",
    password: "",
  });

  const [error, setError] = useState({
    message: "",
  });

  const onInputChange = (e) => {
    const { name, value } = e.target;
    setCredentials({
      ...credentials,
      [name]: value,
    });
  };

  async function processLogin() {
    const db_columns = {
      username: credentials.username,
      pword: credentials.password,
    };

    await api
      .post("/users", db_columns)
      .then(function (response) {
        navigate("/admin/portal");
      })
      .catch(function (error) {
        console.log(error);
        setError({ message: "Invalid username or password" });
      });
  }

  return (
    <div id="login">
      <p className="adminLoginTitle">Admin Login</p>
      {error.message && <p className="err">{error.message}</p>}
      <input
        placeholder="Username"
        id="credBox"
        name="username"
        onChange={onInputChange}
      ></input>
      <input
        placeholder="Password"
        id="credBox"
        type="password"
        name="password"
        onChange={onInputChange}
      ></input>
      <button id="authenticate" onClick={submitLoginForm}>
        LOG IN
      </button>
    </div>
  );
};

export default Login;
