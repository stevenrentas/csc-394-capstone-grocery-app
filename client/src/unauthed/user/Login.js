import React from "react";
import "../../styles/style.css";

const UserLogin = () => {
  return (
    <div id="login">
      <input placeholder="Email Address" id="credBox"></input>
      <input placeholder="Password" id="credBox" type="password" name="password"></input>
      <button id="authenticate">LOG IN</button>
      <p id="otherAuthPage"><a id="otherAuthPage" href={"/signup"}>Create an account</a></p>
    </div>
  );
};

export default UserLogin;
