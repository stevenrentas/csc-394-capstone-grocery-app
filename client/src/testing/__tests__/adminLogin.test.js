import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import Login from '../../unauthed/admin/Login';


const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

describe('Admin Login Components <unauthed/admin/Login.js>', () => {
    
    test('renders components and makes snapshot', () => {
        const {container} = render(<Login/>);
        expect(container).toMatchSnapshot();
    });

    test('Should render admin title,"Admin Login"', () => {

    });

    test('Should render input box for username',() => {
        const { container } = render(<Login/>);
        const userNameInput = container.querySelector('input[name="username"]');
        expect(userNameInput).toBeInDocument;
    });

    test('Should render input box for "password"', () => {
        const { container } = render(<Login/>);
        const passwordInput = container.querySelector('input[name="password"]');
        expect(passwordInput).toBeInDocument;
    });

    test('Should render a "LOG IN" button', () => {
        const {getByText} = render( <Login /> );
        const loginButton = getByText( "LOG IN" );
        expect(loginButton).toBeInDocument;
     });

     test('onClick function called for "LOG IN" button', () => {
        const onClickMock = jest.fn();
        const {getByText} = render( 
            <button onClick={onClickMock}>LOG IN</button>
        );
        const button = getByText('LOG IN');
        fireEvent.click(button);
        expect(onClickMock).toHaveBeenCalled();
    });

    test ('Should render a "Create an account" button',() =>{
        const {getByText} = render(<Login/>);
        const createUserButton = getByText("Create an account");
        expect(createUserButton).toBeInDocument;

    });
});

