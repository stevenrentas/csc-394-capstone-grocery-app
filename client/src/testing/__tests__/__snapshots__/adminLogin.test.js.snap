// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Login Components <unauthed/admin/Login.js> renders components and makes snapshot 1`] = `
<div>
  <div
    id="login"
  >
    <p
      class="adminLoginTitle"
    >
      Admin Login
    </p>
    <input
      id="credBox"
      name="username"
      placeholder="Username"
    />
    <input
      id="credBox"
      name="password"
      placeholder="Password"
      type="password"
    />
    <button
      id="authenticate"
    >
      LOG IN
    </button>
    <p
      id="otherAuthPage"
    >
      <a
        href="/signup"
        id="otherAuthPage"
      >
        Create an account
      </a>
    </p>
  </div>
</div>
`;
